### PR TITLE
[js/webgpu] Allow binary ops with scalar to use the vectorize path

### PR DESCRIPTION
**Pull request by qjia7**
### Description
1. For binary ops, the components is always 4. So the dispatchGroup should be : `{x: Math.ceil(outputSize / 64 /* workgroup size */ / 4 /* component size */)}` instead of `{x: Math.ceil(outputSize / 64 /* workgroup size */ / (vectorize ? 4 : 1) /* vec size */)}`.

2. If any of a or b only has one element, we still can use the vectorize path since the same value will be broadcasted.

### CODE REVIEW

**Comment by qjia7** [comment]
@fs-eire @guschmue Please take a look, thanks.

File: js/web/lib/wasm/jsep/webgpu/ops/binary-op.ts [hunk 1]
```diff
@@ -62,14 +62,24 @@ const createBinaryOpProgramShader =
       let assignment: string;
       if (vectorize) {
         if (doBroadcast) {
-          assignment = `
+          const isAScalar = dimsA.length === 0;
+          const isBScalar = dimsB.length === 0;
```
Comment (fs-eire) [inline js/web/lib/wasm/jsep/webgpu/ops/binary-op.ts:66]: can we extend this from supporting scalar only to supporting 1-sized tensor (eg. dims = [1] , or [1, 1])?

**Review by fs-eire — APPROVED** [review]

**Pull request merged**